src/OrthoTree/OrthoTree.hpp

```diff
@@ -0,0 +1,119 @@
+#include "OrthoTree.h"
+
+namespace ippl {
+
+    template <size_t Dim>
+    OrthoTree<Dim>::OrthoTree(size_t max_depth, size_t max_particles_per_node, const bounds_t& root_bounds)
+        : max_depth_m(max_depth), max_particles_per_node_m(max_particles_per_node),
+        root_bounds_m(root_bounds), morton_helper(max_depth)
+    {
+        std::cout << "Initialized OrthoTree<" << Dim << ">:" << std::endl
+            << "    max_particles: " << max_particles_per_node << std::endl
+            << "    max_depth: " << max_depth
+            << std::endl;
+    }
+
+    template <size_t Dim>
+    void OrthoTree<Dim>::build_tree_naive(particle_t const& particles)
+    {
+        initialize_aid_list(particles);
+
+        std::stack<std::pair<morton_code, size_t>> s;
+        s.push({ morton_code(0), particles.getLocalNum() });
+
+        while ( !s.empty() ) {
+            const auto& [octant, count] = s.top(); s.pop();
+
+            if ( count <= max_particles_per_node_m || morton_helper.get_depth(octant) >= max_depth_m ) {
+                tree_m.push_back(octant);
+                continue;
+            }
+
+            for ( const auto& child_code : morton_helper.get_children(octant) ) {
+                size_t count = 0;
+                for ( const auto& [particle_code, id] : aid_list ) {
+                    if ( morton_helper.is_descendant(particle_code, child_code) ) {
+                        ++count;
+                    }
+                }
+
+                if ( count > 0 ) {
+                    s.push({ child_code, count });
+                }
+            }
+        }
+
+        // if we sort the tree after construction we can compare two trees
+        std::sort(tree_m.begin(), tree_m.end());
+    }
+
+    template <size_t Dim>
+    bool OrthoTree<Dim>::operator==(const OrthoTree& other)
+    {
+        if ( n_particles != other.n_particles ) {
+            return false;
+        }
+
+        if ( tree_m.size() != other.tree_m.size() ) {
+            return false;
+        }
+
+        for ( size_t i = 0; i < tree_m.size(); ++i ) {
+            if ( tree_m[i] != other.tree_m[i] ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    template <size_t Dim>
+    Kokkos::vector<Kokkos::pair<morton_code, Kokkos::vector<size_t>>> OrthoTree<Dim>::get_tree() const
```

Comment:
Discussion topic for later.
Switch to kokkos::view?

Reply:
Added an [issue]( #4 )